source/client/TableView.py

```diff
@@ -3,25 +3,44 @@
 from PodSixNet.Connection import connection, ConnectionListener
 import client.UIConstants as UIC
 from common.Card import Card
+from common.HandAndFoot import Meld_Threshold as Meld_Threshold_HF
+from common.HandAndFoot import wild_numbers as wild_numbers_HF
+from common.Liverpool import Meld_Threshold as Meld_Threshold_LP
+from common.Liverpool import wild_numbers as wild_numbers_LP
+# from common.Liverpool import combineCardDicts as combineCardDicts
 
 
 class TableView(ConnectionListener):
-    """ This displays publicly available info on all the players.
+    """ This displays publicly available info on all the players. """
 
-    It is Hand And Foot Specific
-    """
-
-    def __init__(self, display):
+    def __init__(self, display, ruleset):
         self.display = display
+        self.ruleset = ruleset
+        print('in Tableview: ruleset '+ self.ruleset)
         self.player_names = []
-        self.visible_cards = []
+        self.visible_scards = []        # contains list of serialized cards (from server)
         self.hand_status = []
         self.compressed_info = {}
-        self.playerByPlayer()
+        if self.ruleset == 'Liverpool':
```

Comment:
I apparently didn't put that comment in handView (oops) - but essentially you don't need to check the rulset name if you can actually access the ruleset (which I think you can here).

Reply:
I made changes in HandView (I had a "todo" reminding me to do this).
In TableView it's a bit trickier, because TableView doesn't have controller or state info (as per schematic made early on in this project).    It knows ruleset, but that's just a string.